### PR TITLE
Add var to persist former-agents' roles for SLA metrics

### PR DESCRIPTION
**Pull request by fivetran-jamie**
## PR Overview
**This PR will address the following Issue/Feature:**
https://github.com/fivetran/dbt_zendesk/issues/120

**This PR will result in the following new package version:**

v0.10.1 -- by default nothing will change for folks. they'll need to explicitly set the internal user criteria var 

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

# dbt_zendesk_source v0.10.1
[PR #43](https://github.com/fivetran/dbt_zendesk_source/pull/43) introduces the following updates:

## Feature Updates
- Added the `internal_user_criteria` variable, which can be used to mark internal users whose `USER.role` may have changed from `agent` to `end-user` after they left your organization. This variable accepts SQL that may reference any non-custom field in `USER`, and it will be wrapped in a `case when` statement in the `stg_zendesk__user` model.
  - Example usage:
```yml
# dbt_project.yml
vars:
  zendesk_source:
    internal_user_criteria: "lower(email) like '%@fivetran.com' or external_id = '12345' or name in ('Garrett', 'Alfredo')" # can reference any non-custom field in USER
```
  - Output: In `stg_zendesk__user`, users who match your criteria and have a role of `end-user` will have their role switched to `agent`. This will ensure that downstream SLA metrics are appropriately calculated.

## Under the Hood
- Updated the way we dynamically disable sources. Previously, we used a custom `meta.is_enabled` flag, but, since we added this, dbt-core introduced a native `config.enabled` attribute.  We have opted to use the dbt-native config instead.
- Updated the pull request [templates](/.github).
- Included auto-releaser GitHub Actions workflow to automate future releases.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Will validate internally with @jackiexsun

### CODE REVIEW

File: README.md
Comment (fivetran-jamie) [inline README.md:77]: reminder to myself to add this to the transform README (no release needed there tho) after approval

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie just a few comments around the changes. Once you reply to them and make any updates then let me know for re-review and I can check again and approve.

File: models/stg_zendesk__user.sql
Comment (fivetran-jamie) [reply models/stg_zendesk__user.sql:46]: just ran it on dbt-bigquery==1.3.0 with and without providing `internal_user_criteria: "lower(email) like '%@fivetran.com'"` and it works as expected 😄 

yeah i think it works similarly to how we provide `false` (or true i suppose) as the default value for `using_<possibly missing source table>` variables

File: CHANGELOG.md
Comment (fivetran-jamie) [reply CHANGELOG.md]: added a brief note!

File: README.md
Comment (fivetran-jamie) [reply README.md]: added a little something

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:11]: Small suggestion, but when I ran this exact snippet locally, I got an error that the `=` operator may not be used when comparing STRING to INT. We should probably update this to be consistent that it will work if copy/pasting.

```suggestion
    internal_user_criteria: "lower(email) like '%@fivetran.com' or external_id = '12345' or name in ('Garrett', 'Alfredo')" # can reference any non-custom field in USER
```

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:87]: Same suggestion from CHANGELOG.

```suggestion
    internal_user_criteria: "lower(email) like '%@fivetran.com' or external_id = '12345' or name in ('Garrett', 'Alfredo')" # can reference any non-custom field in USER
```

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie these changes look good to me. I just have one small request to update the example snippet before moving the a release review as I ran into a small issue when trying to copy/paste the snippet provided.

Thanks!

**Pull request merged**
